envoy.dependency.check/envoy/dependency/check/abstract/checker.py

```diff
@@ -195,6 +200,25 @@ async def dep_date_check(
                 "release_dates",
                 [f"Date matches ({dep.release_date}): {dep.id}"])
 
+    async def dep_sha_check(
+            self,
+            dep: "abstract.ADependency") -> None:
+        """Check sha for dependency."""
+        if not await dep.release.sha:
+            self.error(
+                "sha",
+                [f"{dep.id} is a GitHub repository with no inferrable "
+                 "release sha"])
+        elif await dep.release_sha_mismatch:
+            self.error(
+                "sha",
+                [f"SHA mismatch: {dep.id} "
+                 f"{dep.release_sha} != {await dep.release.sha}"])
```

Comment:
i think we just want the first 10 chars of the release_sha for display - not sure if its best to just truncate it here or wheter we should have a property - probs the latter because then you can just reuse/change from one place etc